### ISSUE TITLE
Is g++ 4.8.X supported for building peloton ?

### ISSUE THREAD

**Issue by hemanthmalla** [cmu-db/peloton#864]
Build succeeds with g++ 4.9.4, but fails with  g++ 4.8.4

Building with 4.8.4 throws the following error :
```
/home/ubuntu/src/peloton/src/threadpool/worker.cpp: In constructor ‘peloton::threadpool::WorkerPool::WorkerPool(size_t, peloton::threadpool::TaskQueue*)’:
/home/ubuntu/src/peloton/src/threadpool/worker.cpp:56:38: error: ‘make_unique’ is not a member of ‘std’
     std::unique_ptr<Worker> worker = std::make_unique<Worker>();
                                      ^
/home/ubuntu/src/peloton/src/threadpool/worker.cpp:56:61: error: expected primary-expression before ‘>’ token
     std::unique_ptr<Worker> worker = std::make_unique<Worker>();
                                                             ^
/home/ubuntu/src/peloton/src/threadpool/worker.cpp:56:63: error: expected primary-expression before ‘)’ token
     std::unique_ptr<Worker> worker = std::make_unique<Worker>();
                                                               ^
make[2]: *** [src/CMakeFiles/peloton.dir/threadpool/worker.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/CMakeFiles/peloton.dir/all] Error 2
make: *** [all] Error 2
```

Wiki at [Complier Environment](https://github.com/cmu-db/peloton/wiki/Environment#compiler) states that g++ > 4.7 is supported but [travis builds](https://travis-ci.org/cmu-db/peloton) use g++-4.9 and g++-5

**Comment by crd477** [comment]
On 2017-10-30 14:17, Hemanth Malla wrote:
> Build succeeds with g++ 4.9.4, but fails with g++ 4.8.4
> 
> Building with 4.8.4 throws the following error :
> 
> |/home/ubuntu/src/peloton/src/threadpool/worker.cpp: In constructor 
> ‘peloton::threadpool::WorkerPool::WorkerPool(size_t, 
> peloton::threadpool::TaskQueue*)’: 
> /home/ubuntu/src/peloton/src/threadpool/worker.cpp:56:38: error: 
> ‘make_unique’ is not a member of ‘std’ std::unique_ptr<Worker> worker = 
> std::make_unique<Worker>(); ^ 
> /home/ubuntu/src/peloton/src/threadpool/worker.cpp:56:61: error: 
> expected primary-expression before ‘>’ token std::unique_ptr<Worker> 
> worker = std::make_unique<Worker>(); ^ 
> /home/ubuntu/src/peloton/src/threadpool/worker.cpp:56:63: error: 
> expected primary-expression before ‘)’ token std::unique_ptr<Worker> 
> worker = std::make_unique<Worker>(); ^ make[2]: *** 
> [src/CMakeFiles/peloton.dir/threadpool/worker.cpp.o] Error 1 make[2]: 
> *** Waiting for unfinished jobs.... make[1]: *** 
> [src/CMakeFiles/peloton.dir/all] Error 2 make: *** [all] Error 2 |
> 
> Wiki at Complier Environment 
> <https://github.com/cmu-db/peloton/wiki/Environment#compiler> states 
> that g++ > 4.7 is supported but travis builds 
> <https://travis-ci.org/cmu-db/peloton> use g++-4.9 and g++-5
> 

My nearly-finished Jenkins test setup agrees with this.

The situation with Travis is that although they call their build 
environment "Trusty", they include versions of software that casual 
users cannot install via the Ubuntu package repositories.

-- 
     -Chad

**Comment by apavlo** [comment]
Fix is merged in #865. Thanks to @seojungmin!
